internal/provider/clickhouse_cluster_resource_test.go

```diff
@@ -67,132 +105,101 @@ func TestAccClickhouseClusterResource(t *testing.T) {
 			},
 			// Update and Read testing
 			{
-				Config: testAccClickhouseClusterResourceConfigUpdated(&m2),
+				Config: convertClickHouseModelToHCL(&m2),
 				Check: resource.ComposeAggregateTestCheckFunc(
 					resource.TestCheckResourceAttr(testAccClickhouseId, "name", m2.Name.ValueString()),
-					resource.TestCheckResourceAttr(testAccClickhouseId, "resources.clickhouse.resource_preset_id", "s1-c2-m4"),
+					resource.TestCheckResourceAttr(testAccClickhouseId, "resources.clickhouse.resource_preset_id", "s2-c2-m8"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "resources.clickhouse.disk_size", "51539607552"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "config.log_level", "LOG_LEVEL_TRACE"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "config.max_connections", "120"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "config.kafka.security_protocol", "SASL_SSL"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "config.kafka.sasl_mechanism", "SCRAM_SHA_512"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "config.kafka.sasl_username", "admin"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "config.kafka.sasl_password", "Traffic3-Mushiness-Chariot"),
-					resource.TestCheckResourceAttr(testAccClickhouseId, "config.kafka.enable_ssl_certificate_verification", "true"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "config.kafka.session_timeout_ms", "1m0s"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "access.data_services.0", "transfer"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "access.ipv4_cidr_blocks.1.value", "11.0.0.0/8"),
 					resource.TestCheckResourceAttr(testAccClickhouseId, "access.ipv4_cidr_blocks.1.description", "Office in Cupertino"),
 				),
 			},
+			// Enable autoscaling
+			{
+				Config: convertClickHouseModelToHCL(&m3),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(testAccClickhouseId, "resources.clickhouse.min_resource_preset_id", "s2-c2-m8"),
+					resource.TestCheckResourceAttr(testAccClickhouseId, "resources.clickhouse.max_resource_preset_id", "s2-c4-m16"),
```

Comment:
Could you also check null values of `resource_preset_id` and `disk_size` ?

Reply:
Disk size is not null here, it's diff suppressed